tests/fields/postgres/test_ranges.py

```diff
@@ -283,36 +340,66 @@ def test_query(self):
             )
         ).exists()
 
+    def test_query_single_datetime(self):
+        half_finite_datetime_range = ranges.HalfFiniteDatetimeRange(
+            start=localtime.datetime(2024, 1, 10), end=None
+        )
+        models.HalfFiniteDateTimeRangeModel.objects.create(
+            half_finite_datetime_range=half_finite_datetime_range
+        )
+        assert models.HalfFiniteDateTimeRangeModel.objects.filter(
+            half_finite_datetime_range__contains=half_finite_datetime_range.start
+        ).exists()
+        assert not models.HalfFiniteDateTimeRangeModel.objects.filter(
+            half_finite_datetime_range__contains=half_finite_datetime_range.start
+            - relativedelta.relativedelta(microseconds=1)
+        ).exists()
+
     def test_query_does_not_allow_tuple_values(self):
-        with pytest.raises(TypeError):
+        with pytest.raises(
+            TypeError,
+            match="HalfFiniteDateTimeRangeField may only accept HalfFiniteDateTimeRangeField or datetime objects",
+        ):
             models.HalfFiniteDateTimeRangeModel.objects.filter(
                 half_finite_datetime_range=(
                     localtime.datetime(2024, 1, 10),
                     None,
                 )
             )
-        with pytest.raises(TypeError):
+        with pytest.raises(
+            TypeError,
+            match="HalfFiniteDateTimeRangeField may only accept HalfFiniteDateTimeRangeField or datetime objects",
+        ):
             models.HalfFiniteDateTimeRangeModel.objects.filter(
                 half_finite_datetime_range__overlap=(
                     localtime.datetime(2024, 1, 1),
                     localtime.datetime(2024, 1, 15),
                 )
             )
-        with pytest.raises(TypeError):
+        with pytest.raises(
+            TypeError,
+            match="HalfFiniteDateTimeRangeField may only accept HalfFiniteDateTimeRangeField or datetime objects",
+        ):
```

Comment:
🐼 The use of `match=` is a good addition but perhaps it warrants a separate commit?

Reply:
Okay I'll split this change out.

Reply:
Done